### PR TITLE
fix cookbook

### PR DESCRIPTION
**Pull request by tudorbarascu**

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 1]
```diff
@@ -251,20 +253,18 @@ explains how to do :ref:`modifications with editing buffer <editing-buffer>`.
...
-          layer.setCacheImage(None)
-      else:
-          iface.mapCanvas().refresh()
-
+ ::
 
+  # If caching is enabled, a simple canvas refresh might not be sufficient
+  # to trigger a redraw and you must clear the cached image for the layer
+  if iface.mapCanvas().isCachingEnabled():
+      QgsMapRendererCache().clearCacheImage(layer.id())
```
Comment (tudorbarascu) [inline source/docs/pyqgis_developer_cookbook/vector.rst]: I think this is the new method of clearing the cacheImage. Not 100% sure though.

Comment (ghtmtt) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: I don't know the caching either, but for the rest, +1. Thanks @tudorbarascu

Comment (elpaso) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: IIRC I added that snipped: you should probably check if that is still necessary with QGIS 3 API.

Comment (tudorbarascu) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: @elpaso From what I could grasp, it's not necessary anymore, but I'm not at all sure, so I'm stuck.

Comment (tudorbarascu) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: @nyalldawson Thanks a lot!

**Pull request merged**